packages/chain-adapters/src/evm/EvmBaseAdapter.ts

```diff
@@ -698,4 +703,22 @@ export abstract class EvmBaseAdapter<T extends EvmChainId> implements IChainAdap
   get wsProvider(): unchained.ws.Client<unchained.evm.types.Tx> {
     return this.providers.ws
   }
+
+  async verifySignTx(
+    tx: SignTx<T>,
+    {
+      wallet,
+      accountNumber,
+      receiveAddress,
+    }: GetAddressInput & { receiveAddress: string | undefined },
+  ) {
+    // TODO: simulate the tx to introspect contract interactions for detailed address list
+    const addresses = [
+      await this.getAddress({ wallet, accountNumber }),
+      tx.to,
+      receiveAddress,
+    ].filter((address): address is string => address !== undefined)
+
+    return verify(addresses, tx)
+  }
```

Comment:
this is probably the most fragile of the whole PR, but no worse than whats in develop currently - it requires us to manually pass in the correct addresses. Later we'll want to simulate the tx and verify those to automate it